content/tutorials/reactor-event-cloudflare.textile

```diff
@@ -0,0 +1,256 @@
+</a>
+
+h2. Configuring the game
+
+The first thing we need to do to be able to use the provided code is to create a key.js file in the *./src/* directory. Inside this file you'll export your API key as a variable:
+
+```[js]
+export const ABLY_API_KEY = "YOUR_API_KEY"
+```
+
+Add your own Ably API key inbetween the quotation marks!
+
+You will also need to add the API key into the *./cloudflare-worker/index.js* file. Replace the variable near the top of the file.
+
+h2. Configuring Cloudflare
+
+Configure your Cloudflare Worker as per "these instructions in the Cloudflare Docs":https://developers.cloudflare.com/workers/quickstart/#configure.
+
+During configuration of your Cloudflare worker, you're going edit the *./cloudflare-worker/wrangler.toml* file to include your Cloudflare account_id. (You might be wondering where your Cloudflare API key goes - in order to deploy a Worker using Wrangler, you'll configure Wrangler using it's config subcommand so your API keys never touch the Worker code itself.)
+
+Once you're done with configuration, publish your Worker using Wrangler. 
+
+h2. Configuring Ably Reactor
+
+After publishing your Worker to your Cloudflare account, it will have a URL, you will need this to set up the Ably Reactor.
+
+Go to the dashboard for the Ably app you're using, choose the "Reactor" tab, then click "New Reactor Rule":
+
+<a href="/images/tutorials/reactor/reactor-event-rule-1.png" target="_blank">
+  <img src="/images/tutorials/reactor/reactor-event-rule-1.png" style="width: 100%" alt="Create a new Reactor rule">
+</a>
+
+Choose the "Reactor Event" option:
+
+<a href="/images/tutorials/reactor/reactor-event-rule-2.png" target="_blank">
+  <img src="/images/tutorials/reactor/reactor-event-rule-2.png" style="width: 100%" alt="Select the Reactor Event option">
+</a>
+
+Choose the "Cloudflare Workers" option:
+
+<a href="/images/tutorials/reactor/reactor-event-rule-cloudflare.png" target="_blank">  
+  <img src="/images/tutorials/reactor/reactor-event-rule-cloudflare.png" style="width: 100%" alt="Select the Cloudflare Workers option">
+</a>
+
+Fill out the form to provide Ably with the details it needs to invoke your Cloudflare Worker.
+
+Your Reactor event will now be created, and you'll be able to continue developing the game.
+
+h2. Publishing to an Ably Channel when a delivery is made
+
+Open the file *./src/feature/player/movement.js*
+
+This file contains the React code for handling player movements, but it also contains a connection to our Ably Channel, firstly, on line 18
+
+```[js]
+    window.Ably.connection.on(function(stateChange) {
+        console.log('New connection state is ' + stateChange.current)
+    })
+
+    const outboundChannel = window.Ably.channels.get('cloudflareworkerdemo')
+```
+
+This code establishes a connection to our "cloudflareworkerdemo" channel.
+
+When the player completes a level the `handleWin` function is called, and on line 64 we publish a message to our channel.
```

Comment:
We don't use this syntax in Textile, it's `@`.  Please see https://docs.ably.io/client-lib-development-guide/documentation-formatting-guide/#definition-lists, and manually check the rendering.